client.go

```diff
@@ -0,0 +1,153 @@
+}
+
+// CreateAll calls the endpoint POST /key/:table, executing the statement
+//
+// CREATE type::thing($table, $id) CONTENT $body;
+func (sc Client) CreateAll(table string, body interface{}) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s", table), "POST", body)
+}
+
+// SelectAll calls the endpoint GET /key/:table, executing the statement
+//
+// SELECT * FROM type::table($table);
+func (sc Client) SelectAll(table string) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s", table), "GET", "")
+}
+
+// SelectOne calls the endpoint GET /key/:table/:id, executing the statement
+//
+// SELECT * FROM type::thing(:table, :id);
+func (sc Client) SelectOne(table string, id string) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s/%s", table, id), "GET", "")
+}
+
+// ReplaceOne calls the endpoint PUT /key/:table/:id, executing the statement
+//
+// UPDATE type::thing($table, $id) CONTENT $body;
+func (sc Client) ReplaceOne(table, id string, body interface{}) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s/%s", table, id), "PUT", body)
+}
+
+// UpsertOne calls the endpoint PUT /key/:table/:id, executing the statement
+//
+// UPDATE type::thing($table, $id) MERGE $body;
+func (sc Client) UpsertOne(table, id string, body interface{}) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s/%s", table, id), "PATCH", body)
+}
+
+// DeleteOne calls the endpoint DELETE /key/:table/:id, executing the statement
+//
+// DELETE FROM type::thing($table, $id);
+func (sc Client) DeleteOne(table, id string) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s/%s", table, id), "DELETE", "")
+}
+
+// DeleteAll calls the endpoint DELETE /key/:table/, executing the statement
+//
+// DELETE FROM type::table($table);
+func (sc Client) DeleteAll(table string) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s", table), "DELETE", "")
+}
+
+// Request makes a request to surrealdb to the given endpoint, with the given data. Responses returned from
+// surrealdb vary, and this function will only return the first response
+// TODO: have it return the array, or some other data type that more properly reflects the responses
+func (sc Client) Request(endpoint string, requestType string, body interface{}) (Response, error) {
+	client := &http.Client{}
+	var bodyBytes []byte
+	var err error
+
+	// If it is a string, send it directly though, otherwise try to unmarshal, throwing an error if it fails
+	switch v := body.(type) {
+	case string:
+		bodyBytes = []byte(v)
+	default:
+		bodyBytes, err = json.Marshal(v)
+		if err != nil {
+			return Response{}, err
+		}
+	}
+
+	// TODO: verify its a valid requesttype
+	req, err := http.NewRequest(requestType, sc.URL+endpoint, bytes.NewBuffer(bodyBytes))
+	if err != nil {
+		return Response{}, err
+	}
+	req.Header.Set("NS", sc.NS)
+	req.Header.Set("DB", sc.DB)
+	req.Header.Set("Content-Type", "application/json")
```

Comment:
Hey @MrShutCo this is changing in `1.0.0-beta.8` to `Accept` (which it should always have been).

Can we add an `Accept: application/json` header (in addition), so it works on both `1.0.0-beta.7` and `1.0.0-beta.8`?